lib/index.d.ts

```diff
@@ -1463,14 +1141,14 @@ declare namespace planck {
 
         zero(): Vec2;
         neo(x: number, y: number): Vec2;
-        clone(v: Vec2/*, depricated*/): Vec2;//depricated?
-        isValid(v: any): v is { x: number, y: number };
-        //asssert(o: Vec2): void;
+        clone(v: Vec2): Vec2;
+        isValid(v: any): boolean;
```

Comment:
Why are you removing all typeguards? 😭

Reply:
Can you please explain this a little more, I'm new to TS! :)

Reply:
Of course :) Defining a typeguard basically tells typescript that the function returns a boolean and also some additional information if the returned value is `true`. The Handbook explains it quite nicely: http://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types

But I found a problem: instead of `is` we actually want something like `implements` or `extends` (not possible in TS). I tried `isValid(v: any): v is { x: number, y: number } & v;` but this makes the Typeguard useless.